### PR TITLE
Simplify KeepInterner

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @andrewconner 

We're getting a lot of BatchReadWriteChunk failures. I'm trying to simplify the keep interning process.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala [hunk 1]
```diff
@@ -186,68 +186,39 @@ class KeepInternerImpl @Inject() (
...
-
-        (true, false, improvedKeep)
+    // We intern by (library, uri)
+    val existingKeepOpt = libraryOpt.flatMap { lib => keepRepo.getByUriAndLibrary(uri.id.get, lib.id.get, excludeState = None) }
+
+    val kTitle = List(title.map(_.trim).filter(_.nonEmpty), existingKeepOpt.filter(_.isActive).flatMap(_.title), uri.title).flatten.headOption
+    val kNote = List(note.map(_.trim).filter(_.nonEmpty), existingKeepOpt.filter(_.isActive).flatMap(_.note)).flatten.headOption
+    val keep = Keep(
+      id = existingKeepOpt.map(_.id.get),
+      createdAt = existingKeepOpt.filter(_.isActive).map(_.createdAt).getOrElse(clock.now),
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:196]: I'm not convinced that keeping the old `createdAt` is the correct behavior. Opinions?

**Comment by andrewconner** [comment]
Looks nicer. What are the failures from, though?

**Comment by ryanpbrewster** [comment]
I honestly can't tell. Airbrake says IntegrityConstraintViolation. Looking into it further now.

**Comment by andrewconner** [comment]
Alright. Your refactor looks good though, makes it easier to read.

**Pull request merged**